mithril-explorer/src/components/VerifyCertificate/CertificateVerifier.js

```diff
@@ -0,0 +1,145 @@
+
+import styles from "./styles.module.css";
+
+let nextVerifyEventId = 0;
+
+export const certificateValidationSteps = {
+  ready: 1,
+  validationInProgress: 2,
+  done: 3,
+};
+
+const certificateChainValidationEvents = {
+  started: "CertificateChainValidationStarted",
+  certificateValidated: "CertificateValidated",
+  done: "CertificateChainValidated",
+};
+
+export default function CertificateVerifier({
+  certificateHash,
+  showSpinner = true,
+  onStepChange = (step) => {},
+  onCertificateChange = (certificate) => {},
+}) {
+  const currentAggregator = useSelector((state) => state.settings.selectedAggregator);
+  const [currentStep, setCurrentStep] = useState(certificateValidationSteps.ready);
+  const [certificate, setCertificate] = useState({});
+  const [verificationDuration, setVerificationDuration] = useState(null);
+  const [verificationEvents, setVerificationEvents] = useState([]);
+
+  useEffect(() => {
+    const broadcast_channel = new BroadcastChannel("mithril-client");
+    broadcast_channel.addEventListener("message", clientEventListener);
+
+    return () => broadcast_channel.close();
+  }, []);
+
+  useEffect(() => {
+    onCertificateChange(certificate);
+  }, [certificate, onCertificateChange]);
+
+  useEffect(() => {
+    onStepChange(currentStep);
+  }, [currentStep, onStepChange]);
+
+  useEffect(() => {
+    // Reset state if any
+    setCertificate({});
+    setVerificationEvents([]);
+    setCurrentStep(certificateValidationSteps.ready);
+
+    if (certificateHash) {
+      setCurrentStep(certificateValidationSteps.validationInProgress);
+
+      verifyCertificateChain(currentAggregator, certificateHash)
+        .catch((err) => console.error("Certificate Chain verification error", err))
+        .finally(() => {
+          setCurrentStep(certificateValidationSteps.done);
+        });
+    }
+  }, [currentAggregator, certificateHash]);
+
+  async function verifyCertificateChain(aggregator, certificate_hash) {
+    const genesisVerificationKey = await fetchGenesisVerificationKey(aggregator);
+    const client = new MithrilClient(aggregator, genesisVerificationKey);
+
+    if (certificate_hash !== null && certificate_hash !== undefined) {
+      let startTime = performance.now();
+      const certificate = await client.get_mithril_certificate(certificate_hash);
+      setCertificate(certificate);
+      await client.verify_certificate_chain(certificate_hash);
+      setVerificationDuration(formatProcessDuration(startTime));
+    }
+  }
+
+  function clientEventListener(e) {
+    const event = e.data;
+    let message = <></>;
+
+    switch (event.type) {
+      case certificateChainValidationEvents.started:
+        message = <>The certificate chain validation has started...</>;
+        break;
+      case certificateChainValidationEvents.certificateValidated:
+        message = (
+          <>
+            A certificate has been validated, hash:{" "}
+            <strong>{event.payload.certificate_hash}</strong>
```

Comment:
Maybe we can provide links to the certificate in the explorer (not in this PR :sweat_smile:)?

Reply:
This would means a refactor of how the explorer work to be more in line with what a service like CExplorer does. But yeah, why not :).